src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2067,15 +2070,24 @@ private void ParseUsingDeclaration(in SyntaxListBuilder<RazorSyntaxNode> builder
                 else if (At(CSharpKeyword.Static))
                 {
                     // static using
+                    isStatic = true;
                     AcceptAndMoveNext();
                     AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
+                    nonNamespaceTokenCount = TokenBuilder.Count;
                     TryParseNamespaceOrTypeName(directiveBuilder);
                 }
 
+                var usingStatementTokens = TokenBuilder.ToList().Nodes;
+                var usingContentTokens = usingStatementTokens.Skip(1);
+                var parsedNamespaceTokens = usingStatementTokens
+                    .Skip(nonNamespaceTokenCount)
+                    .Where(s => s.Kind != SyntaxKind.CSharpComment && s.Kind != SyntaxKind.Whitespace && s.Kind != SyntaxKind.NewLine);
+
                 SpanContext.EditHandler.AcceptedCharacters = AcceptedCharactersInternal.AnyExceptNewline;
-                SpanContext.ChunkGenerator = new AddImportChunkGenerator(new LocationTagged<string>(
-                    string.Concat(TokenBuilder.ToList().Nodes.Skip(1).Select(s => s.Content)),
-                    start));
+                SpanContext.ChunkGenerator = new AddImportChunkGenerator(
```

Comment:
Definitely feels odd to be putting all this information on the chunk generator. Any reason to not expose this as part of the node?

Reply:
https://github.com/aspnet/AspNetCore-Tooling/pull/276/files#r261306536